### DISCUSSION TITLE
Github Gist shows large files but not small files "that are this big"

### DISCUSSION THREAD

**Discussion by Zocker1999NET** [community/community#103254] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

I have created a gist, containing small scripts with 2,5 KiB combined and some log/statistic files with 1,7 MiB text (+ README & 2 SVGs). Because of the file names, the bigger files are on the top, while the smaller files are at the bottom.

When opening [the page](https://gist.github.com/Zocker1999NET/8bef576bd385fb05762355e90515402e) (**be aware of potential high CPU usage, esp. with add-ons enabled**), the page fully renders out all the large files but does exactly avoid to render the scripts with the message:

> Sorry about that, but we can’t show files that are this big right now.

I had expected Gist to avoid rendering the large files and maybe render the smaller ones. I would propose to fix this to provide a better experience, especially for users with less powerful computers.

(I cannot attach a screenshot as the page is too long for the screenshot feature of Firefox, it aborts after half of the page.)

**Comment by gil-son** [comment]
GitHub Gist has some limitations on rendering large files directly in the web interface. While text-based files are typically rendered for a quick preview, very large files might not be rendered to prevent performance issues for users.

If your log/statistic files are very large, consider truncating them or including only a sample in the Gist. This can allow users to get a preview without overwhelming the rendering engine

or

you can provide links to the raw content of the large files.

**Reply by Zocker1999NET** [reply]
Thanks for the fast answer, but I think you missed the keypoint (maybe I did not explained it that well):

Gist *is rendering* the LARGE files, bringing Firefox on my machine nearly to its knees with most of my add-ons enabled. It does *not* rendering the smaller files which come afterwards. The bug I see is that the 6 large files are rendered despite their size.

**Comment by bczhc** [comment]
Bump. Another [gist example](https://gist.github.com/bczhc/c3c7981a5e7383ced4325c92b2ad72c7). This makes my markdown note not rendered.

<img width="2164" height="1190" alt="image" src="https://github.com/user-attachments/assets/f4a1d851-49f5-4456-aebf-2e7bd9222432" />

`a.md` and `plot.py` are classified as "big files" however they apparently are not.
